pdk/1.1/modules/ROOT/pages/policies-pdk-configure-features-timer.adoc

```diff
@@ -0,0 +1,161 @@
+----
+
+Add a loop to the `my_async_task` function creates periodic tasks, for example:
+
+[source,rust]
+----
+async fn my_async_task(timer: &Timer, config: &Config) {
+    // While the policy is still running.
+    // Wait for the next cycle.
+    while timer.next_tick().await {
+        // Execute periodic task
+    }
+}
+----
+
+To launch multiple asynchronous tasks, send a reference of the timer to your `async` functions and join them with the `launch` task, for example:
+
+[source,rust]
+----
+let task1 = my_async_task1(&timer, &config);
+let task2 = my_async_task2(&timer, &config);
+let launched = launcher.launch(filter);
+let joined = join!(launched, task1, task2);
+----
+
+NOTE: To view an example policy project that launches asynchronous periodic tasks, see https://github.com/mulesoft/pdk-custom-policy-examples/blob/main/metrics/README.md[Metrics Policy Example^].
+
+== Sync Asynchronous Tasks between Workers
+
+PDK provides the `LockBuilder` to sync workers. The lock ensures that only one worker is completing a task in a time period. Each policy supports multiple locks.
+
+Locks are useful when a policy consumes an HTTP service that has usage limits. A single worker can fetch data and then share the data with other workers to reduce service requests.
+
+To configure a lock:
+
+. Inject the `LockBuilder` into your policy configuration function:
++
+[source,rust]
+----
+#[entrypoint]
+async fn configure(
+    launcher: Launcher,
+    Configuration(bytes): Configuration,
+    clock: Clock,        // Inject the clock to be able to launch async tasks.
+    lock: LockBuilder,   // Inject the lock to be able to synchronize the workers.
+) -> Result<()> {
+----
+
+. Configure the lock and share a reference to it with your asynchronous function:
++
+[source,rust]
+----
+    let lock = lock
+        .new(ID.to_string())
+        .expiration(Duration::from_secs(20))
+        .build();
+    let task = my_async_task(&timer, &lock);
+----
++
+If the asynchronous task that executes `async` function calls, configure the `expiration` of the lock to a value bigger than the total time needed to complete the nested `async` function calls.
+
+. Inside the asynchronous function, acquire the lock before executing the desired task. If no other worker has the lock, the current worker acquires the lock. Otherwise, you must omit the execution of the task and wait for the next task loop, for example:
++
+[source,rust]
+----
+async fn my_async_task(
+    timer: &Timer,
+    lock: &TryLock,
+) {
+   while timer.next_tick().await {
+     if let Some(acquired) = lock.try_lock() {
+       // execute task.
+     }
+   }
+}
+----
++
+After obtaining the lock, if you execute code that requires you ‘await’ a result, you must call the `refresh_lock` function to ensure the lock hasn’t expired. If the lock has expired, the worker is considered unresponsive. Abort the execution of your task as another worker has begun the same task execution
```

Comment:
```suggestion
After obtaining the lock, if you execute code that requires you ‘await’ a result, you must call the `refresh_lock` function to ensure the lock hasn’t expired. If the lock has expired, the worker is considered unresponsive. Abort the execution of your task because another worker is running the same task execution.
```